src/pybind/mgr/dashboard/frontend/src/styles/_carbon-defaults.scss

```diff
@@ -43,7 +43,17 @@ Datatable
 
 tr.cds--expandable-row > td:first-of-type {
   background-color: vv.$white !important;
-  padding-inline-start: 1rem !important;
+}
+
+tr.cds--expandable-row {
+  th,
+  tr td {
+    padding-inline-start: 0;
+  }
+}
+tbody:has(td.cds--table-column-checkbox) > tr.cds--expandable-row[data-child-row] td,
+tr.cds--parent-row.cds--expandable-row + tr[data-child-row] td {
+  padding-inline-start: 1rem;
 }
```

Comment:
I actually don't think it is a good idea to have these overrides. They can easily break with future versions of carbon. 

There's an easier way to achieve this by applying the utility class ```ps-0``` or ```ps-1``` to the table columns in the table component.

```table.component.ts line 396:```
```js 
className: _.isString(col?.cellClass) ? `${col?.cellClass} ps-1` : `${col?.className} ps-1`,
```

Also, I noticed that lots of components pass ```cellClass: 'text-right'``` as a table header prop. This is getting applied currently. But I think the above suggestion overrides it. If this is not what you want you can change the order of the first ```ps-1``` class.

Reply:
thank you. this fixed the issue in a lot of places except the checkbox one, for which I am still using the css overrides.

![Screenshot from 2024-09-16 19-06-11](https://github.com/user-attachments/assets/f8042c53-1d3c-426e-a6f7-712a29f87169)

Reply:
i had to revert the whole thing. I initially thought this was fixed but it only fixed the headers and the body starts misaligning. Due to the urgency I just went for the hack now but once you are back we can improve this.
![Screenshot from 2024-09-17 16-22-00](https://github.com/user-attachments/assets/8701109f-6874-40d3-97f8-6444d3bc6c4b)